### PR TITLE
[DEV APPROVED] 8912 add grouped fields

### PR DESCRIPTION
**Pull request by willhall88**
[TP-8912](https://moneyadviceservice.tpondemand.com/entity/8912-add-grouped-blocks-to-cms-insights)

This addresses multiple blocks for fincap.  
'topics', 'country_of_delivery', 'client_groups, 'data_type'

The main bit of logic and code that this implements is adding a new custom comfy tag: CollectionCheckBoxes

When a new tag is implemented which matches the regex:
`{{ cms:page:IDENTIFIER:collection_check_boxes/PARAM1, PARAM2 }}`

In the form builder the method `collection_check_boxes` will then handle this and build a list of checkboxes, one for each param. 

WHAT THIS PR DOES NOT SOLVE
Unfortunately this PR still saves the checkboxes into one block within the DB. i.e. 
```
#<Comfy::Cms::Block id: 25, 
identifier: "topics", 
content: "---\n- Saving\n- Pensions and Retirement Planning\n", 
...
>
```

### CODE REVIEW

File: features/pages_form/fincap_insight_pages.feature [hunk 1]
```diff
@@ -29,3 +38,16 @@ Feature: Building Insight Pages
...
+      | financial_capability     | checked   |
+      | financial_education      | unchecked |
+      | insurance_and_protection | unchecked |
+      | united_kingdom           | checked   |
+      | usa                      | checked   |
+      | scotland                 | unchecked |
+      | scotland                 | unchecked |
+      | parents_families         | checked   |
+      | children                 | checked   |
+      | young_people             | unchecked |
```
Comment (margOnline) [inline features/pages_form/fincap_insight_pages.feature]: new line

File: features/step_definitions/fincap_pages_steps.rb [hunk 2]
```diff
@@ -50,6 +54,23 @@
   end
 end
 
+When(/^I check$/) do |table|
+  # table is a Cucumber::Core::Ast::DataTable
```
Comment (margOnline) [inline features/step_definitions/fincap_pages_steps.rb]: is this comment needed?

File: lib/cms/form_builder.rb [hunk 3]
```diff
@@ -28,6 +36,29 @@ def page_image(tag, index)
...
+          fieldname: field_name_for(tag)
+        })
+  end
+
+  def page_text(tag, index)
+    default_tag_field(tag, index, :text_area_tag, :data => {'cms-cm-mode' => 'text/html'})
+  end
+
+  def field_name_for(tag)
+    tag.blockable.class.name.demodulize.underscore.gsub(/\//,'_')
```
Comment (margOnline) [inline lib/cms/form_builder.rb]: 🙄

**Comment by willhall88** [comment]
WHAT THIS PR DOES NOT SOLVE
Unfortunately this PR still saves the checkboxes into one block within the DB.

There will be a follow up piece of work that will solve this issue

**Pull request merged**